decentraland/unity-explorer
# Fix: vertical voyager badge

## What does this PR change?
fix #2175

Adjusted logic for the vertical voyager badge. Taking into account
- identity change
- teleportation
- worlds changing
- genesis city beam
- moving platforms

## How to test the changes?

1. Launch the explorer
2. Try to reach 500 meters and wait until you receive the badge. 

Since main logic here is to compare you previous y-position to the new one, several exploits can be done. Check that all these cases handled and doesn't cause accumulating the height :
1. Try genesis city Beam Up several time - it shouldn't accumulate gain
2. Try teleporting between places and between worlds - it shouldn't accumulate gain
3. Try reach something with moving platforms - gain shouldn't accumulate for moving platform, but if you reach somewhere with it, then I think it should count as a gain. 

Try change Identity and verify it resets gain and badge

**NOTE**: We have added a new widget in the Debug Panel called "**_BADGES TRACKING_**" where you will be able to check the current meters reached by your avatar. The value will be WHITE if you didn't reach the target and GREEN if you reached it.

![image](https://github.com/user-attachments/assets/195d0afb-04f8-425b-bc78-2c5003f0ac11)

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
